pkg/resource/deploy/step_generator.go

```diff
@@ -72,15 +69,11 @@ type stepGenerator struct {
 	aliases map[resource.URN]resource.URN
 }
 
-func (sg *stepGenerator) isTargetedUpdate() bool {
-	return sg.updateTargetsOpt.IsConstrained() || sg.replaceTargetsOpt.IsConstrained()
-}
-
```

Comment:
This was incorrect. replaceTargets only indicate that a resource is marked for replace. Targeted updates are different.